### PR TITLE
Fix hierarchical package view

### PR DESCRIPTION
**Pull request by Flanker32**
Fix issues with hierarchical package view and typos

### CODE REVIEW

File: src/views/dataNode.ts [hunk 1]
```diff
@@ -35,8 +35,9 @@ export abstract class DataNode extends ExplorerNode {
     public async revealPaths(paths: INodeData[]): Promise<DataNode> {
         const childNodeData = paths.shift();
         const childs: ExplorerNode[] = await this.getChildren();
```
Comment (andxu) [inline src/views/dataNode.ts]: children

File: src/views/hierarchicalPackageRootNode.ts [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
-            child instanceof HierachicalPackageNode && hierachicalNodeData.name.startsWith(child.nodeData.name));
-        return childNode === null ? null : childNode.revealPaths(paths);
+        const hierarchicalNodeData = paths[0];
+        const children: ExplorerNode[] = await this.getChildren();
+        const childNode = <DataNode>children.find((child: DataNode) =>
+            hierarchicalNodeData.name.startsWith(child.nodeData.name + ".") || hierarchicalNodeData.name === child.nodeData.name);
+        // don't shift when child node is an hierarchical node, or it may lose data of package node
+        if (!(childNode instanceof HierarchicalPackageNode)) {
+            paths.shift();
+        }
```
Comment (andxu) [inline src/views/hierarchicalPackageRootNode.ts]: (childNode && paths.length > 0) ? childNode.revealPaths(paths) : childNode;

File: src/views/dependencyDataProvider.ts [hunk 3]
```diff
@@ -71,9 +71,9 @@ export class DependencyDataProvider implements TreeDataProvider<ExplorerNode> {
     public async revealPaths(paths: INodeData[]): Promise<DataNode> {
         const projectNodeData = paths.shift();
         const projects = await this.getRootProjects();
-        const correspondProject = <DataNode>projects.find((node: DataNode) =>
-            node.path === projectNodeData.path && node.nodeData.name === projectNodeData.name);
-        return correspondProject.revealPaths(paths);
+        const correspondProject = projects ? <DataNode>projects.find((node: DataNode) =>
```
Comment (andxu) [inline src/views/dependencyDataProvider.ts]: correspond is VI, please change it to ADJ.

Comment (yaohaizh) [reply src/views/dependencyDataProvider.ts]: The correspond prefix is not necessary. Every variable is correspond variable in the code.

**Pull request merged**